### PR TITLE
Upgrade jscodeshift to 0.5.0 and all other deps

### PR DESCRIPTION
**Pull request by dfederm**
**What kind of change does this PR introduce?**
Dependencies upgrade

**Did you add tests for your changes?**
Nothing to add. Adjusted a snapshot as needed though.

**If relevant, did you update the documentation?**

**Summary**
* Upgrade to the latest jscodeshift
* Upgrade all other dependencies while I was here too since it was trivial (except for yargs)

**Does this PR introduce a breaking change?**
No.

### CODE REVIEW

**Comment by ematipico** [comment]
Hi @dfederm, would you mind to do a rebase from master? We just a merged a PR with some changes to the `package.json` and now there are conflicts

File: lib/init/transformations/top-scope/__snapshots__/top-scope.test.js.snap [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 exports[`top-scope transforms correctly using "top-scope-0" data 1`] = `
 "var test = 'me';
-module.exports = {}"
+module.exports = {}
```
Comment (montogeek) [inline lib/init/transformations/top-scope/__snapshots__/top-scope.test.js.snap]: Such a strange change from Jest

File: package.json [hunk 2]
```diff
@@ -99,7 +99,8 @@
     "p-lazy": "^1.0.0",
     "prettier": "^1.5.3",
     "resolve-cwd": "^2.0.0",
-    "supports-color": "^5.2.0",
+    "supports-color": "^5.3.0",
+    "uglifyjs-webpack-plugin": "^1.2.2",
```
Comment (ematipico) [inline package.json]: Is this new? :O

File: package.json [hunk 3]
```diff
@@ -110,16 +111,16 @@
     "webpack": "^4.0.0-beta.1"
```
Comment (ematipico) [inline package.json]: Could you update this to be `^4.0.0`?

File: package.json (hunk 2 above)
Comment (dhruvdutt) [reply package.json]: Ah! Almost nobody noticed it. :laughing: :eyes:

Comment (ematipico) [reply package.json]: We should remove it

Comment (dfederm) [reply package.json]: Woops, a bad rebase on my end. It got removed in #309 and I merged badly. Will remove.

**Pull request merged**